python/pyspark/serializers.py

```diff
@@ -223,12 +224,13 @@ def _create_batch(series):
     # If a nullable integer series has been promoted to floating point with NaNs, need to cast
     # NOTE: this is not necessary with Arrow >= 0.7
     def cast_series(s, t):
-        if t is None or s.dtype == t.to_pandas_dtype():
+        if t is None or s.dtype == t.to_pandas_dtype() or type(t) == pa.TimestampType:
```

Comment:
for timestamps, pyarrow.DataType.to_pandas_dtype will be a "datetime64[us]" while s.dtype is "datetime64[ns]" so they are not equal but trying to use `astype` will give an error.  So I think this is fine.

Reply:
Here `TimestampType` was removed from the pyarrow namespace since 0.7.0 but I opened a JIRA to add it back

https://issues.apache.org/jira/browse/ARROW-1683

We created a new `pyarrow.types` API which should replace these checks with 

`pa.types.is_timestamp(t)` 

but that requires Arrow 0.8.0. I would recommend making this transition before Spark 2.3.0. Timeline for Arrow 0.8.0 is early November

Reply:
Add a to-do here.